loadfile/loadfile_test.go

```diff
@@ -21,21 +21,20 @@ func Test_LoadContext(t *testing.T) {
 	assert.NoError(t, os.MkdirAll(testdir, os.ModePerm))
 
 	// create a directory
-	dirname := "mydir"
-	dirpath := filepath.Join(testdir, dirname)
-	assert.NoError(t, os.MkdirAll(dirpath, os.ModePerm))
+	dirPrefix := "mydir"
+	dirPath, err := os.MkdirTemp(testdir, dirPrefix+"*")
```

Comment:
I read the documentation again, and I realize that the `+"*"` actually isn't meaningful here -- the default behavior is to add a random suffix to the pattern string you provide, and you only need the explicit "*" if you want the randomized string somewhere else in the name, like `"path/*-foo"`. So this isn't wrong, but it could be simplified a little. 😬 

And a thought ... are "we" (e.g., the Go test system) deleting these temporary files (directories) automatically, e.g., by deleting the entire `TestDir` tree? Or should the test be deferring a cleanup function?